### ISSUE TITLE
@keyframes not hoisted out of @media, even when using @at-root

### ISSUE THREAD

**Issue by duke9509** [sass/dart-sass#1079]
SCSS:
```scss
@media (prefers-reduced-motion: no-preference) {
    .my-class {
        position: absolute;
        animation: test 1s infinite;
        
        @keyframes test {
            from {
                top: 0;
            }
            
            to {
                top: 10px;
            } 
        }
    }
}
```

Expected output:
```css
@media (prefers-reduced-motion: no-preference) {
  .my-class {
    position: absolute;
    animation: test 1s infinite;
  }
}

@keyframes test {
  from {
    top: 0;
  }
  to {
    top: 10px;
  }
}
```

Actual output (Dart Sass 1.26.10):
```css
@media (prefers-reduced-motion: no-preference) {
  .my-class {
    position: absolute;
    animation: test 1s infinite;
  }
  @keyframes test {
    from {
      top: 0;
    }
    to {
      top: 10px;
    }
  }
}
```

Wrapping `@keyframes` in `@at-root` has no effect.

This is arguably a minor issue, but it would be nice to be able to always put the keyframes right next to their use without having to worry about if there's a media query involved (which there often will be with the advent of `prefers-reduced-motion`).

Is this expected behavior? Or should the `@keyframes` block be hoisted?

**Comment by jathak** [comment]
Sorry for the delay in responding to this!

This is expected behavior, as rules nested within a `@media` query generally shouldn't be hoisted out of them (as doing so would break the encapsulation of the `@media` query), so nested `@keyframes` rules are only hoisted out of style rules, not `@media` queries.

By default, `@at-root` similarly only hoists nested rules out of style rules. However, you can use [`@at-root (without: media)` to bypass `@media` queries as well](https://sass-lang.com/documentation/at-rules/at-root#beyond-style-rules).


This should yield the results you're looking for:

```scss
@media (prefers-reduced-motion: no-preference) {
  .my-class {
    position: absolute;
    animation: test 1s infinite;

    @at-root (without: media) {
      @keyframes test {
        from {
          top: 0;
        }

        to {
          top: 10px;
        } 
      }
    }
  }
}
```

**Comment by duke9509** [comment]
No worries! Thanks for getting back to me!

I was surprised to hear you say this was expected, but doing some more research it looks like my knowledge was a bit out of date. I was not aware the CSS grammar had been updated to allow nested at-rules. So it looks like I filed the bug in the wrong place! The nested keyframes were breaking my media queries in Safari, so I guess I should file a WebKit bug instead, haha.

Thanks for the tip about `@at-root`; the parameter is new to me! I will start using that until Safari handles nested at-rules better!
